pkg/zk/generators.go

```diff
@@ -226,7 +226,8 @@ func makeZkEnvConfigString(z *v1beta1.ZookeeperCluster) string {
 		"LEADER_PORT=" + strconv.Itoa(int(ports.Leader)) + "\n" +
 		"CLIENT_HOST=" + z.GetClientServiceName() + "\n" +
 		"CLIENT_PORT=" + strconv.Itoa(int(ports.Client)) + "\n" +
-		"CLUSTER_NAME=" + z.GetName() + "\n"
+		"CLUSTER_NAME=" + z.GetName() + "\n" +
+		"CLUSTER_SIZE=" + fmt.Sprint(z.Spec.Replicas) + "\n"
```

Comment:
Also update the CLUSTER_SIZE variable in config map on scale up or scale down ( when ever user updates the `zk.spec.replicas` field so that if user triggers an upgrade after doing a scale up/scale down on a zookeeper cluster ( with version 0.2.5) the correct CLUSTER_SIZE is maintained and teardown works as expected.

Reply:
This is taken care by `reConcileConfigmap`. I have tried an upgrade from `0.2.5-rc0` to `0.2.6-rc0` after scaling replicas and it is working fine.